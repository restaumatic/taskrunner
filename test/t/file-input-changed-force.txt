# no toplevel

echo lol > input.txt
git init -q
git add input.txt

go() {
  taskrunner -n mytask "$@" bash -e -c '
    snapshot input.txt
    echo "Expensive computation, input is: $(cat input.txt)"
    tr a-z A-Z < input.txt > output.txt
  '
}

go

echo "I am changed now" > input.txt
go --force

# Going back to original input should result in re-running!
echo lol > input.txt
go

# TODO: switch to actual assertions for cases like this
echo "This should be LOL: $(cat output.txt)"
