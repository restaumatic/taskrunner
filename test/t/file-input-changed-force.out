-- output:
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation, input is: lol
[mytask] info | success
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation, input is: I am changed now
[mytask] info | success
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation, input is: lol
[mytask] info | success
This should be LOL: LOL
